nucliadb/nucliadb/ingest/orm/node.py

```diff
@@ -101,9 +101,11 @@ async def create_shard_by_kbid(
 
         try:
             # When choosing the nodes, we need to exclude nodes in which there is already a shard from the same KB
-            kb_nodes = [
-                replica.node for shard in kb_shards.shards for replica in shard.replicas
-            ]
+            # XXX removed, why do we care about this? How was this working before?
+            # kb_nodes = [
+            #     replica.node for shard in kb_shards.shards for replica in shard.replicas
+            # ]
```

Comment:
The idea was to spread the replicas of logic shards of a KB as much as possible.

If we are OK of having replicas of different logic shards assigned to the same node, then we can remove it.

Reply:
Okay, I think I'll tweak this logic. I think right now we might be in a situation where we wouldn't find enough nodes when we did have them available